### PR TITLE
pls: Unveil search paths with browse permissions

### PR DESCRIPTION
**Pull request by bcoles**
Recent changes to `unveil` broke `pls`. This PR fixes it.

pls

### CODE REVIEW

File: Userland/Utilities/pls.cpp [hunk 1]
```diff
@@ -28,40 +28,46 @@ bool unveil_paths(const char*);
...
-    bool did_unveil_ok = false;
+    auto trusted_path = "/bin:/usr/bin:/usr/local/bin";
+
+    // Unveil all trusted paths with browse permissions
+    for (auto directory : String { trusted_path }.split(':')) {
+        if (unveil(directory.characters(), "b") < 0) {
+            perror("unveil");
+            return 1;
+        }
+    }
```
Comment (gunnarbeutner) [inline Userland/Utilities/pls.cpp:39]:
```suggestion
    // Unveil all trusted paths with browse permissions
    auto trusted_directories = Array { "/bin", "/usr/bin", "/usr/local/bin" };
    for (auto directory : trusted_directories) {
        if (unveil(directory, "b") < 0) {
            perror("unveil");
            return false;
        }
    }
```

Comment (bcoles) [reply Userland/Utilities/pls.cpp]: done. i had this originally defined at the top of the file. `trusted_paths` should eventually be abstracted away into a config flag.

Comment (gunnarbeutner) [reply Userland/Utilities/pls.cpp]: Are you sure? Your latest commit shows the old code for me. :)

Comment (bcoles) [reply Userland/Utilities/pls.cpp]: Fixed. I didn't notice you had also changed the directory parsing to an `Array`.

An `Array` makes more sense, for now.

File: Userland/Utilities/pls.cpp [hunk 2]
```diff
@@ -28,40 +28,45 @@ bool unveil_paths(const char*);
...
 {
-    bool did_unveil_ok = false;
+    // Unveil all trusted paths with browse permissions
+    auto trusted_directories = Array { "/bin", "/usr/bin", "/usr/local/bin" };
+    for (auto directory : trusted_directories) {
+        if (unveil(directory, "b") < 0) {
+            perror("unveil");
+            return false;
+        }
+    }
```
Comment (linusg) [inline Userland/Utilities/pls.cpp:38]: not for this PR, just curious: this means that `pls` will not find binaries in custom `PATH` directories right now as those are never `unveil()`'d with browse permissions, correct?

Comment (bcoles) [reply Userland/Utilities/pls.cpp:38]: Correct. The user must provide the full path to the executable if the executable resides outside the trusted system paths.

**Pull request merged**